bot/cogs/queue.py

```diff
@@ -178,78 +161,65 @@ async def join(self, ctx):
+                        all_readied = await self.start_match(ctx, queue)
                     except asyncio.TimeoutError:
                         return
 
                     if all_readied:
-                        queue.bursted = queue.active  # Save bursted queue for player draft
-                        queue.active = []  # Reset the player queue to empty
+                        await self.bot.db_helper.delete_queued_users(ctx.guild, *queue)
 
                     return
 
-        # Send message based on outcome
-        embed = self.queue_embed(ctx.guild, title)
+        embed = await self.queue_embed(ctx.guild, title)
 
         # Delete last queue message
-        if queue.last_msg:
-            try:
-                await queue.last_msg.delete()
-            except discord.errors.NotFound:
-                pass
-
-        queue.last_msg = await ctx.send(embed=embed)
+        await self.update_last_msg(ctx, embed)
 
     @commands.command(brief='Leave the queue (or the bursted queue)')
     async def leave(self, ctx):
         """ Check if the member can be remobed from the guild and remove them if so. """
-        queue = self.guild_queues[ctx.guild]
+        removed = await self.bot.db_helper.delete_queued_users(ctx.guild, ctx.author)
 
-        if ctx.author in queue.active:
-            queue.active.remove(ctx.author)
+        if ctx.author.id in removed:
             title = f'**{ctx.author.display_name}** has been removed from the queue '
         else:
             title = f'**{ctx.author.display_name}** isn\'t in the queue '
 
-        embed = self.queue_embed(ctx.guild, title)
-
-        if queue.last_msg:
-            try:
-                await queue.last_msg.delete()
-            except discord.errors.NotFound:
-                pass
+        embed = await self.queue_embed(ctx.guild, title)
 
-        queue.last_msg = await ctx.channel.send(embed=embed)
+        # Update queue display message
+        await self.update_last_msg(ctx, embed)
 
     @commands.command(brief='Display who is currently in the queue')
     async def view(self, ctx):
-        """  Display the queue as an embed list of mentioned names. """
-        queue = self.guild_queues[ctx.guild]
-        embed = self.queue_embed(ctx.guild, 'Players in queue for 10-mans')
-
-        if queue.last_msg:
-            try:
-                await queue.last_msg.delete()
-            except discord.errors.NotFound:
-                pass
+        """ Display the queue as an embed list of mentioned names. """
+        title = 'Players in queue for 10-mans'
+        embed = await self.queue_embed(ctx.guild, title)
 
-        queue.last_msg = await ctx.send(embed=embed)
+        # Update queue display message
+        await self.update_last_msg(ctx, embed)
 
     @commands.command(usage='remove <user mention>',
                       brief='Remove the mentioned user from the queue (must have server kick perms)')
```

Comment:
I think tying this to server kick permissions could be annoying for some people, could we look at making this configurable in the future?

Reply:
We could yeah. I figure server kick is good cause it's practically the same thing as removing from the queue. We could just make all elevated commands admin-only if you think the consistency would be better?

Reply:
> We could yeah. I figured server kick is good cause it's practically the same thing as removing from the queue. We could just make all elevated commands admin-only if you think the consistency would be better?

Yeah that could be better, kick perms work for now just one to look at in the future